### ISSUE TITLE
Incorrect auto-correction

### ISSUE THREAD

**Issue by tomhughes** [bbatsov/rubocop#2654]
When run on this code:

```ruby
netmask = (~IPAddr.new(address).mask(0)).mask(prefix)
```

rubocop reports:

```
cookbooks/networking/recipes/default.rb:38:56: C: Don't use parentheses around a method call.
```

and corrects it to:

```ruby
netmask = ~IPAddr.new(address).mask(0).mask(prefix)
```

but removing the parentheses means than the `~` is applied after the call to the `mask` method instead of before it.

This is a change between 0.34.2 and 0.36.0 as the earlier version didn't complain about this code.

**Comment by bbatsov** [comment]
Seems we failed to account for unary ops in this cop.

**Comment by lumeet** [comment]
Ouch, I'll take care of this.
